Cabal/Distribution/System.hs

```diff
@@ -122,7 +122,7 @@ osAliases Permissive FreeBSD = ["kfreebsdgnu"]
 osAliases Compat     FreeBSD = ["kfreebsdgnu"]
 osAliases Permissive Solaris = ["solaris2"]
 osAliases Compat     Solaris = ["solaris2"]
-osAliases _          Android = ["linux-android"]
+osAliases _          Android = ["linux-android", "linux-androideabi" ]
```

Comment:
I think we should gate these based on `cabal-version` / `CabalSpecVersion`. This affects cabal file format.

cc @hvr

Reply:
I think in practice this is more likely to fix old packages than break them. `linux-android` was added without such a gate https://github.com/haskell/cabal/pull/5199/files, and before that GHC broke things on Cabal's end when it got rid of its notion of ios and android, so way back when `os(android)` probably worked for both OSes.

Reply:
The fact that we were sloppy in the past doesn't justify being sloppy in the future.

```
Warning: The following errors will cause portability problems on other
environments:
Warning: Unknown operating system name 'linux-androideabi'
Warning: Hackage would reject this package.
```

The `osAliases` are part of *cabal file format* and I'm going to be the strict about their changes. Let's do things properly, even it requires more extensive changes. The more we do it so, the easier future changes will be.

---

I think the **right** solution is to have separate `osAliases`: the one used in `.cabal` files, and the ones used to match against `System.Info.os`.

- `.cabal` files list should not be changed easily, and should be documented. We really don't need be able to use `mingw32` to mean `windows`
- latter ones can be changed on need.

Many purposes, tricky to change. Decouple: profit!

Reply:
Yeah decoupling sound good. I only expect anyone to use anything but `os(android)`. I wouldn't even spend any time advocating anything else to work.

`os(linux-android)` was just something people started using as a workaround to `android` not being recognized -- a sort of workdaround that I don't even think works anymore as `os(android) || os(linux-androideabi)` didn't work when I tried it. In fact, I would recommend you revision that away on Hackage if there's any up there.

Reply:
Also, does `ClassificationStrictness` already handle this split, albiet obliquely?